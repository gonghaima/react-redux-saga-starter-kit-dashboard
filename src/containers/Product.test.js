import React from "react";
import { shallow, render, mount } from "enzyme";
import ProductContainer, { Product } from "./Product";

const mockDispatch = jest.fn();

const store = {
  dispatch: mockDispatch,
  product: {
    data: []
  },
  selection: {
    selectionItems: [4, 8, 10, 20, 40]
  }
};
const props = {
  dispatch: mockDispatch,
  product: {
    data: []
  },
  selection: {
    selectionItems: [4, 8, 10, 20, 40]
  }
};

const shallowWithStore = (component, store) => {
  const context = {
    store
  };
  return shallow(component, { context });
};

function setupProuctContainer(ownProps = props) {
  return shallow(<ProductContainer {...ownProps} />, {
    attachTo: document.getElementById("react")
  });
}

function setupProuct() {
  return shallow(<Product {...props} />, {
    attachTo: document.getElementById("react")
  });
}

function renderProuct() {
  return render(<Product {...props} />, {
    attachTo: document.getElementById("react")
  });
}
function mountProuct() {
  return mount(<Product {...props} />, {
    attachTo: document.getElementById("react")
  });
}

function setupProductWithStore() {
  return shallowWithStore(<Product {...props} />, store);
}

describe("ProductContainer", () => {
  const wrapper = setupProuctContainer();
  wrapper.setProps({ dispatch: mockDispatch });
  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Product", () => {
  const wrapper = setupProuct();
  wrapper.setProps({ dispatch: mockDispatch });
  it("should render properly", () => {
    expect(mockDispatch).toHaveBeenCalled();
  });
});
