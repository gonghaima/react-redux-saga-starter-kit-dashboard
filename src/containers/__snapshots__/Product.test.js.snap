// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductContainer should render properly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
